lib/thrift/CMakeLists.txt

```diff
@@ -30,3 +31,14 @@ zephyr_library_sources(
   ${THRIFT_UPSTREAM}/lib/cpp/src/thrift/TApplicationException.cpp
   ${THRIFT_UPSTREAM}/lib/cpp/src/thrift/TOutput.cpp
 )
+
+zephyr_library_sources_ifdef(CONFIG_THRIFT_ZLIB
```

Comment:
Looks like this is missing `_TRANSPORT` at the end

Also, you can likely combine these two `zlib_library_sources_ifdef()` blocks into one.